### PR TITLE
osd/scrub: only call publish_stats_to_osd() where allowed and needed

### PR DESCRIPTION
**Pull request by ronen-fr**
This patch removes several unnecessary calls to publish_stats_to_osd() in the clear_pgscrub_state pathway used during interval changes. We didn't actually have anything to publish and we can't validly call publish_stats_to_osd() when we aren't an active primary anyway (which we can't be during an interval change).

Fixes: https://tracker.ceph.com/issues/58496

### CODE REVIEW

**Comment by ronen-fr** [comment]
This PR contains one of the two fixes (one of the two commits) in
https://github.com/ceph/ceph/pull/50088
which approval is delayed.

**Comment by ljflores** [comment]
jenkins test make check

**Review by neha-ojha — APPROVED** [review]
LGTM

**Comment by rzarzynski** [comment]
It looks we haven't backported the problematic commit:

```
$ git log origin/quincy | grep 1295b3752c8305cb57f69676beb22de07d575e01 || echo no\ commit
no commit
```

File: src/osd/scrubber/pg_scrubber.cc [hunk 1]
```diff
@@ -537,6 +537,16 @@ void PgScrubber::on_primary_change(
     << dendl;
 }
 
+/*
+ * A note re the call to publish_stats_to_osd() below:
```
Comment (rzarzynski) [inline src/osd/scrubber/pg_scrubber.cc:541]: How about rephrasing this lline? Run into a _parser error_.

**Review by rzarzynski — APPROVED** [review]
LGTM per https://tracker.ceph.com/issues/58496#note-11.

File: src/osd/scrubber/pg_scrubber.cc (hunk 1 above)
Comment (ronen-fr) [reply src/osd/scrubber/pg_scrubber.cc:541]: @rzarzynski - not sure I understand the remark?

Comment (rzarzynski) [reply src/osd/scrubber/pg_scrubber.cc:541]: Nothing big -- today it was pretty parseable. I think last time it was the combination of singular (_a note_), _re_ and lack of verb before _below_.

**Review by athanatos — APPROVED** [review]

**Pull request merged**